### PR TITLE
InputText externalValidation Prop and Ring Component

### PR DESCRIPTION
**Pull request by zohlinger-hw**
- Created a new Ring component for UI Feedback
- Implemented Ring in InputText with use of new externalValidation Prop
- Added Stories for new InputText functionality and Ring component

### CODE REVIEW

File: src/Ring/index.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+const Ring = props => {
+  const { className, size } = props
+
+  const circles = [...Array(4)].map((_, index) => {
+    return (
+      <RingCircle
+        key={index}
+        style={{
+          borderColor: `${getThemeVariable('color')(props)} transparent transparent transparent`,
```
Comment (joesaraceno) [inline src/Ring/index.js:47]: only the top border is every non-transparent?

File: src/Ring/index.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+          width: size * 0.8,
+          height: size * 0.8,
```
Comment (joesaraceno) [inline src/Ring/index.js:49]: why is size default to 80 if you are doing match on it always?
should size default prop be whatever makes these not need math?

**Review by joesaraceno — APPROVED** [review]

File: src/Ring/index.js (hunk 2 above)
Comment (zohlinger-hw) [reply src/Ring/index.js:49]: The RingRoot takes the full size value and the RingCircles all get sized slightly smaller than the RingRoot. I tried to pull these into the styled div but haven't been able to get it to work right

Comment (joesaraceno) [reply src/Ring/index.js:49]: ah. maybe try something like `width: ${props => props.size}`

File: src/Ring/index.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+const RingRoot = styled.div`
+  display: inline-block;
+  position: relative;
+  width: ${props => props.size}px;
+  height: ${props => props.size}px;
+`
+
+const Ring = props => {
+  const { className, size } = props
+  console.log('color: ', getThemeVariable('color')(props))
```
Comment (joesaraceno) [inline src/Ring/index.js:48]: can probably remove the logs now that you got the props figured out?

**Review by joesaraceno — APPROVED** [review]

File: src/Ring/index.js (hunk 3 above)
Comment (zohlinger-hw) [reply src/Ring/index.js:48]: Good catch!

**Review by joesaraceno — APPROVED** [review]

**Pull request merged**